CMakeLists.txt

```diff
@@ -320,10 +320,14 @@ if (WITH_SYSTEM_PROVIDED_3PARTY)
   set(GFLAGS_USE_TARGET_NAMESPACE ON)
   find_package(gflags REQUIRED)
 
+  find_package(PkgConfig)
+  pkg_check_modules(jansson REQUIRED IMPORTED_TARGET jansson)
+  add_library(jansson::jansson ALIAS PkgConfig::jansson)
+
   find_package(expat CONFIG REQUIRED)
-  find_package(jansson CONFIG REQUIRED)
   find_package(pugixml REQUIRED)
   find_package(utf8cpp REQUIRED)
+  add_library(utf8cpp::utf8cpp ALIAS utf8cpp)
```

Comment:
This is dependent on the library and which target name it provides (with a namespace or without one). Actually, this is version dependent for utf8cpp:
- utf8cpp version 4.0.5 seems to provide `utf8cpp::utf8cpp` [on its own](https://github.com/nemtrif/utfcpp/blob/v4.0.5/utf8cppConfig.cmake.in#L7)
- utf8cpp version 3.2.5 [does not](https://github.com/nemtrif/utfcpp/blob/v3.2.5/utf8cppConfig.cmake.in#L7)

Gentoo provides utf8cpp version 3.2.1. It might work without this commit for version 4.0.5. If you are not willing to merge it, I would provide it within the Gentoo tooling, since it is distribution specific then.

Reply:
This is the hell that I was afraid of when using system packages instead of bundled ones. Upcoming changes to the code will require utfcpp 4+, older version won't work.

Reply:
Not really. This is the reason, why distribution specific patches exist. Just make OrganicMaps work for the version you also provide as a 3party module (hopefully the newest one) and the distribution maintainer must take care in case the distribution provides only an older version.

In this case, I was not aware of the differences until writing the above comment and Gentoo clearly provides a too old version, so I would say, this should be a distribution specific patch. However, I cannot test against utf8cpp 4.0.5.

Reply:
I have tested compiling against utf8cpp 4.0.5 and it builds find without this patch. Therefore, I already dropped it.

Reply:
Note: both versions provide `utf8::cpp` as an alias for whatever target it exports (namespaced or not).

Reply:
Hmm, thanks for the hint, but the Gentoo version (3.2.1) does not even provide `utf8::cpp`. I'll leave it as it is.